clayui.com/src/components/Editor/index.js

```diff
@@ -54,39 +55,24 @@ const Editor = ({
 		];
 	}
 
-	function useStickyState(defaultValue, key) {
-		const inBrowser = typeof window !== 'undefined';
+	const inBrowser = typeof window !== 'undefined';
 
-		const localStorage = {
-			getItem(key) {
-				return inBrowser ? window.localStorage.getItem(key) : null;
-			},
-
-			setItem(key, value) {
-				if (inBrowser) {
-					window.localStorage.setItem(key, value);
-				}
-			},
-		};
-
-		const [value, setValue] = React.useState(() => {
-			const stickyValue = localStorage.getItem(key);
-
-			return stickyValue !== null
-				? JSON.parse(stickyValue)
-				: defaultValue;
-		});
-
-		React.useEffect(() => {
-			localStorage.setItem(key, JSON.stringify(value));
-		}, [key, value]);
+	const localStorage = {
+		getItem(key) {
+			return inBrowser ? window.localStorage.getItem(key) : null;
+		},
 
-		return [value, setValue];
-	}
+		setItem(key, value) {
+			if (inBrowser) {
+				window.localStorage.setItem(key, value);
+			}
+		},
+	};
 
 	const [collapseCode, setCollapseCode] = useStickyState(
 		true,
-		'collapse-code'
+		'collapse-code',
+		localStorage
```

Comment:
Having to pass in `localStorage` makes the hook less convenient to use now. If another file wants to use it, they will import it, and run into exactly the problem that you ran into that led you to write the `inBrowser` stuff and the `localStorage` wrapper. I think you should move all of _that_ too.

Counter-argument: users can now pass in an arbitrary object that conforms to the local storage interface. Such flexibility!

Counter-counter-argument:

![squinting](https://user-images.githubusercontent.com/7074/88394347-97f3b400-cdbf-11ea-84c3-07d03401fcca.gif)

One useful rule of thumb about whether this kind of thing should be in another file in the first place: I don't extract something that is used in exactly one place into a separate file unless the place where I am extracting it from is getting large and unwieldy (not really the case here?). Keeping things private until they need to be public gives you more freedom to change them, which is a good thing; because hard-to-change abstractions are The Enemy™ and easy-to-change abstractions are The Best™.

PS. Enjoy your vacations @kresimir-coko.

Reply:
Alrighty @wincent that comment sounds like a fork in the road that just meets back into a single road at the other side.

I'll go ahead and add the `inBrowser` & `localStorage` to the exported hook. 

The reason why I didn't do it initially was that I anticipated that `const inBrowser = typeof window !== 'undefined';` wouldn't execute fast enough if it was imported through a separate file, and that we didn't want to declare `localStorage` whenever the hook was being called.

P.S. Vacation was great, the weather wasn't 100% but we managed 🏖️